cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/NextRun/index.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import NamespaceStore from 'services/NamespaceStore';
+import {MyPipelineApi} from 'api/pipeline';
+import IconSVG from 'components/IconSVG';
+import {Observable} from 'rxjs/Observable';
+import Duration from 'components/Duration';
+
+export default class NextRun extends Component {
+  static propTypes = {
+    pipelineInfo: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+  }
+
+  componentWillMount() {
+    if (this.props.pipelineInfo.type === 'Realtime') { return; }
+
+    // Interval only runs after the delay, so have to
+    // initiall call the function first
+    this.getNextRun();
+    this.interval = Observable.interval(30 * 1000)
+      .subscribe(this.getNextRun.bind(this));
+  }
+
+  componentWillUnmount() {
+    if (this.interval && this.interval.unsubscribe) {
+      this.interval.unsubscribe();
+    }
+  }
+
+  state = {
+    loading: true,
+    nextRun: null
+  };
+
+  getNextRun() {
+    let namespace = NamespaceStore.getState().selectedNamespace;
```

Comment:
nit: we added `getCurrentNamespace()` helper function to NamespaceStore, so you can just use that